foodfuncs.py

```diff
@@ -0,0 +1,404 @@
+    #     if conn:
+    #         conn.close()
+    #         print("the connection is closed.")
+
+def is_ingredient_duplicate(name):
+    # WORKS
+    """input: name of ingredient that user is trying to insert
+        output: true if duplicate, false if not
+    """
+    # reads inglist, loops through entries and separates out ingredient names to compare to name
+    file = open("inglist.txt")
+    ingredients = file.read().split("\n")
+    file.close()
+    for item in ingredients: 
+        n = item.split(", ")
+        if n[1] == name:
+            return True
+    return False
+    # try:
+    #     conn = create_connection("recipes.db")
+    #     cur = conn.cursor()
+    #     dup = cur.execute("SELECT id FROM ingredients WHERE ingredient_name = ?", (name,),).fetchall()
+    #     if dup:
+    #         return True
+    #     else:
+    #         return False
+    # except sqlite3.Error as error:
+    #     print("There was an error: ", error)
+    # finally:
+    #     if conn:
+    #         conn.close()
+    #         print("the connection is closed.")
+
+def greatest_ids():
+    # DOESN'T WORK
+    """ output: tuple of recipe id and ingredient id integers
+    """
+    reclist = open("reclist.txt")
+    inglist = open("inglist.txt")
+    rec = reclist.read().split("\n")
+    ing = inglist.read().split("\n")
+    if rec != [] and ing != []:
+        # loop through the list, split each item and return it to the spot 
+        for index, i in enumerate(ing):
+            i = tuple(i.split(", "))
+            ing[index] = i
+        for ind, r in enumerate(rec):
+            r = tuple(r.split(", "))
+            rec[ind] = i 
+        return int(rec.pop()[0]), int(ing.pop()[0])
+    else: 
+        return 0, 0
+    # try:
+    #     conn = create_connection("recipes.db")
+    #     cur = conn.cursor()
+    #     rec = cur.execute("SELECT MAX(id) FROM recipes").fetchall()
+    #     ing = cur.execute("SELECT MAX(id) FROM ingredients").fetchall()
+    #     if type(rec[0][0]) == int and type(ing[0][0]) == int:
+    #         recipe = rec[0][0]
+    #         ingredient = ing[0][0]
+    #     else:
+    #         recipe = 0
+    #         ingredient = 0
+    #     cur.close()
+    #     if conn: 
+    #         conn.close()
+    #     return recipe, ingredient
+    # except sqlite3.Error as error:
+    #     print(f"Could not connect and find max values: {error}")
+
+
+def add_new_recipe(recipe_name, instructions, ingredients):
+    """
+    input: string recipe_name, string instructions, dictionary of 
+    ingredients with associated quantities
+    Checks for duplicate recipe and ingredients with is_duplicate_?
+    assigns an id to recipe and each ingredient that is at least 
+    one more than the greatest id using greatest_id
+    Adds respective data points to the db
+    """
+    inglist = []
+    ingreclist = []
+    # finds the last id or 0 if table is empty, then add 1 to provide the next id
+    if greatest_ids():
```

Comment:
check out try, except statements

```suggestion
    try:
        start_ids = greatest_ids()
        rec = start_ids[0] + 1
        ing = start_ids[1] + 1
    except:
        raise ValueError("Something went wrong with greatest ids.")
```